sktime/forecasting/base/_base.py

```diff
@@ -260,6 +266,7 @@ def fit(self, y, X=None, fh=None):
         self._is_vectorized = vectorization_needed
         # we call the ordinary _fit if no looping/vectorization needed
         if not vectorization_needed:
+            fh = self._update_fh_freq(fh=fh, index=y_inner.index)
```

Comment:
could we instead move it to `_check_fh` which gets an additional argument, `y_inner`?

Reply:
I had the same idea and I tried myself. But `sktime.forecasting.base._base.BaseForecaster.predict_residuals` stops me (for reference, to reproduce run `pytest --no-cov sktime/forecasting/tests/test_all_forecasters.py::TestAllForecasters::test_predict_residuals[DirRecTabularRegressionForecaster-fh=1-int-int-True]`). This method something strange with `fh` and I can not unravel it.

Reply:
can you push a branch where that breaks? It does not break on this branch or on `main`, does it?

Reply:
I have pushed one more commit. I have managed to do what you asked. Hopefully there are no side effects... I run only tests that are most likely to fail with this change and they pass.

The usual problem, the longer you refactor, the higher the probability that the code gives up :D